File: rust/src/models/background_task_type.rs
// this file is @generated
use std::fmt;

use serde::{
    Deserialize,
    Serialize,
};

#[derive(
    Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize,
)]
pub enum BackgroundTaskType {
    #[default]
    #[serde(rename = "endpoint.replay")]
    EndpointReplay,
    #[serde(rename = "endpoint.recover")]
    EndpointRecover,
    #[serde(rename = "application.stats")]
    ApplicationStats,
    #[serde(rename = "message.broadcast")]
    MessageBroadcast,
    #[serde(rename = "sdk.generate")]
    SdkGenerate,
    #[serde(rename = "event-type.aggregate")]
    EventTypeAggregate,
    #[serde(rename = "application.purge_content")]
    ApplicationPurgeContent,
}

impl fmt::Display for BackgroundTaskType {
    fn fmt(
        &self,
        f: &mut fmt::Formatter,
    ) -> fmt::Result {
        let value = match self {
            Self::EndpointReplay => "endpoint.replay",
            Self::EndpointRecover => "endpoint.recover",
            Self::ApplicationStats => "application.stats",
            Self::MessageBroadcast => "message.broadcast",
            Self::SdkGenerate => "sdk.generate",
            Self::EventTypeAggregate => "event-type.aggregate",
            Self::ApplicationPurgeContent => "application.purge_content",
        };
        f.write_str(value)
    }
}

File: php/src/Models/BackgroundTaskType.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

enum BackgroundTaskType: string implements \JsonSerializable
{
    case ENDPOINT_REPLAY = 'endpoint.replay';
    case ENDPOINT_RECOVER = 'endpoint.recover';
    case APPLICATION_STATS = 'application.stats';
    case MESSAGE_BROADCAST = 'message.broadcast';
    case SDK_GENERATE = 'sdk.generate';
    case EVENT_TYPE_AGGREGATE = 'event-type.aggregate';
    case APPLICATION_PURGE_CONTENT = 'application.purge_content';

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return self::from($data);
    }

    public function jsonSerialize(): mixed
    {
        return $this->value;
    }
}
